### PR TITLE
bpo-46522: fix concurrent.futures and io AttributeError messages

### PR DESCRIPTION
**Pull request by graingert**
https://bugs.python.org/issue46522

### CODE REVIEW

**Comment by graingert** [comment]
other examples of module dunder AttributeError:

https://github.com/python/cpython/blob/99b0a6de215c8e37c412d3a786e4d872096ab05b/Lib/zoneinfo/__init__.py#L27
https://github.com/python/cpython/blob/ee60550e9ba3ab94ca43a890cf4313b63ffa1a81/Lib/urllib/parse.py#L793
https://github.com/python/cpython/blob/ee60550e9ba3ab94ca43a890cf4313b63ffa1a81/Lib/unittest/__init__.py#L94

this one doesn't use `{name!r}` but `'{name}'`:
https://github.com/python/cpython/blob/ee60550e9ba3ab94ca43a890cf4313b63ffa1a81/Lib/sqlite3/__init__.py#L71

**Review by MaxwellDupre — APPROVED** [review]
Ran all tests and got:
`test_ioctl skipped -- Unable to open /dev/tty`
But I note that there's a bug for this: Issue39813

Also unrelated: `test_ossaudiodev` failed

**Review by asvetlov — APPROVED** [review]

**Comment by asvetlov** [comment]
The error message text is changed. While the text is not a part of stable API and can be wildly changed, we usually don't do it in bugfix releases without a strong reason. 
That's why I don't want to backport the PR.

File: Misc/NEWS.d/next/Library/2022-01-25-15-31-04.bpo-46522.tYAlX4.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+fix concurrent.futures and io AttributeError messages
```
Comment (asvetlov) [inline Misc/NEWS.d/next/Library/2022-01-25-15-31-04.bpo-46522.tYAlX4.rst]: Hmm, the NEWs is not clear.
Please describe better what is changed.

**Review by asvetlov — CHANGES_REQUESTED** [review]

File: Misc/NEWS.d/next/Library/2022-01-25-15-31-04.bpo-46522.tYAlX4.rst (hunk 1 above)
Comment (graingert) [reply Misc/NEWS.d/next/Library/2022-01-25-15-31-04.bpo-46522.tYAlX4.rst]:
```suggestion
Make various module `__getattr__` AttributeErrors more closely match a typical AttributeError
```

**Comment by graingert** [comment]
I have made the requested changes; please review again

**Review by asvetlov — APPROVED** [review]

**Comment by asvetlov** [comment]
thanks!

**Pull request merged**